### PR TITLE
Additional checks

### LINKED ISSUE #29: Additional checks on Schnorr and VRF Pks and VRF Proof
In particular:

- The keyverify() functions should also verify that the point (the pk) is not the point at infinity;
- When deserializing and/or receiving a VRF proof, we should check that the gamma point is different from zero and perform the group membership check;
- Other checks ?

### PR DESCRIPTION
**Pull request by DanieleDiBenedetto**
Closes #29

### CODE REVIEW

File: api/Cargo.toml [hunk 1]
```diff
@@ -7,9 +7,9 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-algebra = {version = "0.1.0", git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
-primitives = {version = "0.1.0", features = ["merkle_tree", "signature", "vrf"], git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
-proof-systems = {version = "0.1.0", features = ["groth16"], git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
+algebra = {version = "0.1.0", git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "read_checked_unchecked"}
+primitives = {version = "0.1.0", features = ["merkle_tree", "signature", "vrf"], git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "read_checked_unchecked"}
```
Comment (DanieleDiBenedetto) [inline api/Cargo.toml:11]: Change to  GingerLib's development branch, when corresponding PR in GingerLib will be approved and merged in development branch.

File: demo-circuit/Cargo.toml [hunk 2]
```diff
@@ -6,12 +6,12 @@ edition = "2018"
 include = ["Cargo.toml", "src"]
 
 [dependencies]
-algebra = {version = "0.1.0", git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
-primitives = {version = "0.1.0", features = ["signature"], git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
-r1cs-crypto = {version = "0.1.0", features = ["signature"], git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
-r1cs-core = {version = "0.1.0", git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
-r1cs-std = {version = "0.1.0", git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
-proof-systems = {version = "0.1.0", features = ["groth16"], git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
+algebra = {version = "0.1.0", git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "read_checked_unchecked"}
```
Comment (DanieleDiBenedetto) [inline demo-circuit/Cargo.toml:9]: Change to GingerLib's development branch, when corresponding PR in GingerLib will be approved and merged in development branch.

File: ouroboros/Cargo.toml [hunk 3]
```diff
@@ -5,7 +5,7 @@ authors = ["DanieleDiBenedetto <daniele@horizenlabs.io>"]
 edition = "2018"
 
 [dependencies]
-algebra = {version = "0.1.0", git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
-primitives = {version = "0.1.0", features = ["vrf"], git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "development"}
+algebra = {version = "0.1.0", git = "https://github.com/ZencashOfficial/ginger-lib.git", branch = "read_checked_unchecked"}
```
Comment (DanieleDiBenedetto) [inline ouroboros/Cargo.toml:8]: Change to GingerLib's development branch, when corresponding PR in GingerLib will be approved and merged in development branch.

File: ouroboros/src/vrf/mod.rs [hunk 4]
```diff
@@ -14,6 +14,7 @@ use primitives::{crh::{
     },
 }};
 use rand::rngs::OsRng;
+use primitives::vrf::ecvrf::FieldBasedEcVrfPk;
 
 #[derive(Clone)]
```
Comment (albertog78) [inline ouroboros/src/vrf/mod.rs:19]: comment the whole package until we will define better the needed apis

**Review by albertog78 — APPROVED** [review]

**Review by albertog78 — APPROVED** [review]

**Review by albertog78 — APPROVED** [review]

**Pull request merged**